src/utils/xhr-loader.ts

```diff
@@ -191,21 +192,37 @@ class XhrLoader implements Loader<LoaderContext> {
             };
 
             this.callbacks.onSuccess(response, stats, context, xhr);
-          }
+          };
 
           if (context.responseType === 'arraybuffer') {
-            console.log('Response headers', xhr.getAllResponseHeaders());
+            /**
+             * Creates a map with all the response headers from the request
+             */
+            const headers = xhr.getAllResponseHeaders();
+            const headersArray = headers.trim().split(/[\r\n]+/);
+            const headersMap = {};
+            headersArray.forEach(function (line) {
+              const parts = line.split(': ');
+              const header: any = parts.shift();
+              const value = parts.join(': ');
+              headersMap[header] = value;
+            });
+
             const vividas = new Vividas({
-              streamType: 'VOD'
+              // eslint-disable-next-line
+              streamType: STREAM_TYPE || 'VOD',
             });
 
             // data = xhr.response;
             // len = data.byteLength;
-            vividas.decrypt(xhr.response).then((decryptedData => {
-              resolve(decryptedData, decryptedData.byteLength);
-            })).catch((error) => {
-              console.log('Error decrypting segment', error);
-            });
+            vividas
+              .decrypt(xhr.response, headersMap)
+              .then((decryptedData) => {
+                resolve(decryptedData, decryptedData.byteLength);
+              })
+              .catch((error) => {
+                console.log('Error decrypting segment', error);
```

Comment:
I would suggest using `console#error` instead to be more explicit.

Reply:
Sure! Done!